python-package/xgboost/spark.py

```diff
@@ -0,0 +1,225 @@
+#
+# Copyright (c) 2022 by Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark import keyword_only
+from pyspark.ml.common import inherit_doc
+
+from xgboost.ml.dmlc.param import _XGBoostClassifierBase, _XGBoostClassificationModelBase
```

Comment:
I think it's more appropriate to use `xgboost.pyspark.PySparkXGBClassifier` instead of this java style module.

Reply:
From the convention of PySpark, we should keep the same API name for PySpark and JVM.
See https://github.com/apache/spark/blob/branch-3.2/mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala#L58 and https://github.com/apache/spark/blob/branch-3.2/python/pyspark/ml/classification.py#L1952